servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpSetCookie.java

```diff
@@ -248,7 +248,18 @@ static HttpSetCookie parseSetCookie(final CharSequence setCookieString, boolean
                             break;
                     }
                     parseState = ParseState.Unknown;
-                    i += 2;
+                    if (validateContent) {
+                        if (i >= setCookieString.length() || ' ' != setCookieString.charAt(i)) {
+                            throw new IllegalArgumentException(
+                                    "a space is required after ; in cookie attribute-value lists");
```

Comment:
Multiple other tests fail with this exception. Example:
```
DefaultHttpSetCookiesTest > decodeDuplicateNamesRO() FAILED
    java.lang.IllegalArgumentException: a space is required after ; in cookie attribute-value lists
        at io.servicetalk.http.api.DefaultHttpSetCookie.parseSetCookie(DefaultHttpSetCookie.java:253)
        at io.servicetalk.http.api.HeaderUtils.parseSetCookie(HeaderUtils.java:364)
        at io.servicetalk.http.api.ReadOnlyHttpHeaders.getSetCookiesIterator(ReadOnlyHttpHeaders.java:295)
        at io.servicetalk.http.api.DefaultHttpSetCookiesTest.decodeDuplicateNames(DefaultHttpSetCookiesTest.java:60)
        at io.servicetalk.http.api.DefaultHttpSetCookiesTest.decodeDuplicateNamesRO(DefaultHttpSetCookiesTest.java:56)
```
See build logs for full info.

Reply:
This was probably caused by my attempt to move the improvements over bit-by-bit. I have updated the PR to contain all of the improvements in one go.